diff --git a/python/cudnn/__init__.py b/python/cudnn/__init__.py
index c152d0c..c8fc4a6 100644
--- a/python/cudnn/__init__.py
+++ b/python/cudnn/__init__.py
@@ -155,19 +155,7 @@ pygraph.execute_plan_at_index = _execute_plan_at_index
 
 
 def load_cudnn():
-    # First look at python site packages
-    lib_path = glob.glob(
-        os.path.join(sysconfig.get_path("purelib"), "nvidia/cudnn/bin/cudnn64_9.dll")
-    )
-
-    if lib_path:
-        assert (
-            len(lib_path) == 1
-        ), f"Found {len(lib_path)} libcudnn.dll.x in nvidia-cudnn-cuXX."
-        lib = ctypes.windll.LoadLibrary(lib_path[0])
-    else:  # Fallback
-        lib = ctypes.windll.LoadLibrary("cudnn64_9.dll")
-
+    lib = ctypes.CDLL(@store-cudnn.so-path@)
     handle = ctypes.cast(lib._handle, ctypes.c_void_p).value
     _pybind_module._set_dlhandle_cudnn(handle)
 
