diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -7,7 +7,7 @@
 set(CMAKE_POLICY_VERSION_MINIMUM 3.18)
 set(BUILD_MOCK OFF)
 
-option(CUDNN_FRONTEND_USE_SYSTEM_DLPACK "Whether dlpack should use the system version or fetched." OFF)
+option(CUDNN_FRONTEND_USE_SYSTEM_DLPACK "Whether dlpack should use the system version or fetched." ON)
 
 IF(CUDNN_FRONTEND_USE_SYSTEM_DLPACK)
   find_package(dlpack REQUIRED)
