diff --git a/python/cudnn/__init__.py b/python/cudnn/__init__.py
--- a/python/cudnn/__init__.py
+++ b/python/cudnn/__init__.py
@@ -52,9 +52,9 @@ def _dlopen_cudnn():
-    else:  # Fallback
-        try:
-            lib = ctypes.CDLL("libcudnn.so.9")
-        except Exception:
+    else:  # Fallback
+        try:
+            lib = ctypes.CDLL(@store-cudnn.so-path@)
+        except Exception:
             try:
                 lib = ctypes.CDLL("libcudnn.so")
             except Exception:
