diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 3f7824c..478c45e 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -2,16 +2,8 @@ cmake_minimum_required(VERSION 3.18)
 
 Include(FetchContent)
 
-# Fetch and build dlpack
-set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
-set(CMAKE_POLICY_VERSION_MINIMUM 3.18)
-set(BUILD_MOCK OFF)
-FetchContent_Declare(
-  dlpack
-  GIT_REPOSITORY https://github.com/dmlc/dlpack
-  GIT_TAG        v1.1
-)
-FetchContent_MakeAvailable(dlpack)
+# Find dlpack
+find_package(dlpack CONFIG REQUIRED)
 
 # Find python
 find_package(Python 3.8 COMPONENTS Interpreter Development.Module REQUIRED)
@@ -64,7 +56,7 @@ target_compile_definitions(_compiled_module PRIVATE NV_CUDNN_FRONTEND_USE_DYNAMI
 target_link_libraries(
     _compiled_module
     
-    PRIVATE dlpack
+    PRIVATE dlpack::dlpack
 )
 
 if(MSVC)
