diff --git i/python/cudnn/__init__.py w/python/cudnn/__init__.py
index 28ed2c2..88b9285 100644
--- i/python/cudnn/__init__.py
+++ w/python/cudnn/__init__.py
@@ -181,30 +181,7 @@ def load_cudnn():
 
 
 def _dlopen_cudnn():
-    # First look at python site packages
-    lib_path = glob.glob(
-        os.path.join(
-            sysconfig.get_path("purelib"), "nvidia/cudnn/lib/libcudnn.so.*[0-9]"
-        )
-    )
-
-    if not lib_path:
-        lib_path = glob.glob(
-            os.path.join(
-                sysconfig.get_path("purelib"), "nvidia/cudnn_jit/lib/libcudnn.so.*[0-9]"
-            )
-        )
-
-    if lib_path:
-        assert (
-            len(lib_path) == 1
-        ), f"Found {len(lib_path)} libcudnn.so.x in nvidia-cudnn-cuXX."
-        lib = ctypes.CDLL(lib_path[0])
-    else:  # Fallback
-        try:
-            lib = ctypes.CDLL("libcudnn.so.9")
-        except Exception:
-            lib = ctypes.CDLL("libcudnn.so")
+    lib = ctypes.CDLL(@store-cudnn.so-path@)
 
     handle = ctypes.cast(lib._handle, ctypes.c_void_p).value
     _pybind_module._set_dlhandle_cudnn(handle)
