From 88694e862380b33a4cbb81a2a30f65b7cdc62b6d Mon Sep 17 00:00:00 2001
From: Ayan Das <bvits@riseup.net>
Date: Tue, 16 Jul 2024 21:09:58 +0530
Subject: [PATCH] Make PeachPy processes deterministic

---
 peachpy/name.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/peachpy/name.py b/peachpy/name.py
index b6a03dc..412079d 100644
--- a/peachpy/name.py
+++ b/peachpy/name.py
@@ -86,13 +86,13 @@ class Namespace:
                 self.names[scope_name.name] = scope
         else:
             assert scope_name.name is None
-            self.prenames.setdefault(scope_name.prename, set())
+            self.prenames.setdefault(scope_name.prename, {})
             if subscoped_name:
                 for subscope in iter(self.prenames[scope_name.prename]):
                     if isinstance(subscope, Namespace) and subscope.scope_name is scope_name:
                         subscope.add_scoped_name(subscoped_name)
                         return
-            self.prenames[scope_name.prename].add(scope)
+            self.prenames[scope_name.prename][scope] = None
 
     def assign_names(self):
         # Step 1: assign names to symbols with prenames with no conflicts
-- 
2.45.2

